scipy/stats/tests/test_multivariate.py

```diff
@@ -2592,6 +2592,44 @@ def test_entropy_vs_numerical_integration(self, df, cov, ref, tol):
         mvt = stats.multivariate_t(loc, cov, df)
         assert_allclose(mvt.entropy(), ref, rtol=tol)
 
+    @pytest.mark.parametrize(
+        "df, dim, ref, tol",
+        [
+            (10, 1, 1.5212624929756808, 1e-15),
+            (100, 1, 1.4289633653182439, 1e-13),
+            (500, 1, 1.420939531869349, 1e-14),
+            (1e20, 1, 1.4189385332046727, 1e-15),
+            (1e100, 1, 1.4189385332046727, 1e-15),
+            (10, 10, 15.069150450832911, 1e-15),
+            (1000, 10, 14.19936546446673, 1e-13),
+            (1e20, 10, 14.189385332046728, 1e-15),
+            (1e100, 10, 14.189385332046728, 1e-15),
+            (10, 100, 148.28902883192654, 1e-15),
+            (1000, 100, 141.99155538003762, 1e-14),
+            (1e20, 100, 141.8938533204673, 1e-15),
+            (1e100, 100, 141.8938533204673, 1e-15),
+        ]
+    )
+    def test_extreme_entropy(self, df, dim, ref, tol):
+        # Reference values were calculated with mpmath:
+        # from mpmath import mp
+        # mp.dps = 500
+        #
+        # def mul_t_mpmath_entropy(dim, df=1):
+        #     dim = mp.mpf(dim)
+        #     df = mp.mpf(df)
+        #     halfsum = 0.5 * (dim + df)
+        #     half_df = 0.5 * df
+        #
+        #     return float(
+        #         -mp.loggamma(halfsum) + mp.loggamma(half_df)
+        #         + 0.5 * dim * mp.log(df * mp.pi)
+        #         + halfsum * (mp.digamma(halfsum) - mp.digamma(half_df))
+        #         + 0.0
```

Comment:
Please add one test that confirms that the shape term is implemented correctly. That is, add a test with nontrivial covariance matrix that substantially affects the value of the entropy.

For instance, a simpler (but less accurate) approach to this PR would have been to use the fact that the `multivariate_t` entropy approaches the `multivariate_normal` entropy as the number of degrees of freedom gets bigger. Your test could simply compare the two entropies with a nontrivial covariance matrix and large degrees of freedom.

Then, you could confirm that the implementation of the shape term in the non-asymptotic formula is correct by comparing the `multivariate_t.entropy` with slightly different `df` - on either side of the threshold that switches between them. They should be nearly the same.